From 358a4d144ab6f7b48d1dd17f80fce78f3bd94117 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Thu, 02 Jan 2020 17:27:33 +0200
Subject: [PATCH] [SW-6739] arcboot: libsb: use rsa fixed

To improve execution speed of the signature verification
we are using now fixed exponent to speed up decryption of
the signature.

The fixed exponent has pre-calculated tables to speed up decryption,
this comes at the cost of allowing key sizes in 64bit increment instead
of 8bit increment with standard rsa. as we only use 3072bit keys this
has no effect expect speeding up the decryption of the signature.

16KB data on pldm with hashing of data blob -
rsa_standard: 37 mintues
rsa_fixed: 3 mintues

Change-Id: Ic919dac4831e8d028ed487cecc6f4f06a42b943a
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/Makefile b/Makefile
index a018d5b..9f2401b 100644
--- a/Makefile
+++ b/Makefile
@@ -45,7 +45,7 @@
 	@$(MAKE) arcboot.bin arcboot_rom.bin
 
 libsb libellipsys:
-	@$(MAKE) -C sdk $@ SB_SIGN_MODE=rsa_std > /dev/null
+	@+$(MAKE) -C sdk $@ SB_SIGN_MODE=rsa_fixed > /dev/null
 sbtool:
 	@$(MAKE) -C sdk $@ > /dev/null
 %.bin: %
